aesara/tensor/elemwise.py

```diff
@@ -1404,34 +1377,29 @@ def perform(self, node, inp, out):
 
         variable = np.array(input, dtype=acc_dtype)
 
-        if axis:
-            # Reducing functions built using np.frompyfunc() do not
-            # support reduction along multiple axes. Hence loop through
-            # each, otherwise numpy's inbuilt reduction functions
-            # support reduction along multiple axes directly.
-            if self.ufunc_is_vectorized:
-                to_reduce = reversed(sorted(axis))
-                for dimension in to_reduce:
-                    variable = self.ufunc.reduce(variable, dimension, dtype=acc_dtype)
-            else:
-                variable = self.ufunc.reduce(variable, axis=tuple(axis))
-            output[0] = _asarray(variable, dtype=node.outputs[0].type.dtype)
+        # Reducing functions built using np.frompyfunc() do not
+        # support reduction along multiple axes. Hence loop through
+        # each, otherwise numpy's inbuilt reduction functions
+        # support reduction along multiple axes directly.
+        if self.ufunc_is_vectorized:
+            to_reduce = reversed(sorted(axis))
+            for dimension in to_reduce:
+                variable = self.ufunc.reduce(variable, dimension, dtype=acc_dtype)
         else:
-            # Force a copy
-            output[0] = np.array(variable, copy=True, dtype=node.outputs[0].type.dtype)
+            variable = self.ufunc.reduce(variable, axis=tuple(axis))
+        output[0] = _asarray(variable, dtype=node.outputs[0].type.dtype)
 
     def infer_shape(self, fgraph, node, shapes):
-        (ishape,) = shapes
-        axis = self.axis
-        if axis is None:
+        (
+            ishape,
+            axis_shape,
+        ) = shapes
+        if axis_shape == 0:
             return ((),)
-        return (
-            [
-                ishape[i]
-                for (i, b) in enumerate(node.inputs[0].type.broadcastable)
-                if i not in axis
-            ],
+        final_shape = (
+            [ishape[i] for (i, b) in enumerate(node.inputs[0].type.broadcastable)],
```

Comment:
The biggest issue with this approach occurs over here. That is, we don't know the contents of `axis` yet. Therefore when the shape is inferred it equals shape of input itself (for now).